staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -106,7 +106,12 @@ func lookupServiceName(name string) (EgressType, error) {
 }
 
 func tunnelHTTPConnect(proxyConn net.Conn, proxyAddress, addr string) (net.Conn, error) {
-	fmt.Fprintf(proxyConn, "CONNECT %s HTTP/1.1\r\nHost: %s\r\n\r\n", addr, "127.0.0.1")
```

Comment:
After some googling (including finding the spec https://tools.ietf.org/html/rfc7231#section-4.3.6), I definitely believe that 127.0.0.1 is wrong, but it's not completely obvious that repeating the address is correct?

@caesarxuchao @Jefftree @cheftako do you know why after proxying we telling the destination we think we're reaching them as localhost?

Reply:
In golang http lib, when setup the proxy, it also use the CONNECT method, and the host is same as the request addr, ref: https://github.com/golang/go/blob/master/src/net/http/transport.go#L1670

And it use req.Write()to write the request. https://github.com/golang/go/blob/master/src/net/http/transport.go#L1694

Reply:
Maybe the value of Host depend on how the server uses it :)